docker/usr/local/package/intel-graphics-compiler.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+# Exit immediately on error
+set -e
+
+# Get our method as the name of this script, url and args
+URL="$1"
+shift
+
+# Set installation prefix, default to /usr/local unless an external power says otherwise
+PREFIX=${PREFIX:-"/usr/local"}
+
+BUILD_FOLDER="/var/tmp/build"
+
+# Pull in the toolchain arguments
+. /usr/local/toolchain.sh
+
+# Setup the temporary build directories
+mkdir -p "${BUILD_FOLDER}"
+cd "${BUILD_FOLDER}"
+
+# Download the source code
+download-and-extract "${URL}"
+
+echo "Configuring using cmake"
+
+# Find the closest configure file to the root
+CMAKELISTS_FILE=$(find -type f -name 'CMakeLists.txt' -printf '%d\t%P\n' | sort -nk1 | cut -f2- | head -n 1)
+cd $(dirname "${CMAKELISTS_FILE}")
+
+echo "Configuring using cmake file ${CMAKELISTS_FILE}"
+
+# Clone the dependency
+git clone https://github.com/intel/vc-intrinsics vc-intrinsics
+
+# Do an out of source build
+mkdir -p build
+cd build
+
+# Configure using cmake
+cmake .. -GNinja \
+    -DCMAKE_BUILD_TYPE="Release" \
+    -DCMAKE_TOOLCHAIN_FILE="${PREFIX}/toolchain.cmake" \
+    -DCMAKE_INSTALL_PREFIX=/usr \
+    -DCMAKE_INSTALL_LIBDIR=lib \
+    -DIGC_OPTION__ARCHITECTURE_TARGET='Linux64' \
+    -DIGC_PREFERRED_LLVM_VERSION='11.0.0' \
+    -DVC_INTRINSICS_SRC="../../vc-intrinsics" \
+    -DINSTALL_SPIRVDLL=0 \
+    -DINSTALL_GENX_IR=ON \
+    -Wno-dev
+
+# Build the package
+ninja -j `nproc`
```

Comment:
Never use `command` style when executing commands. use $(command) instead
```suggestion
ninja -j $(nproc)
```

Reply:
is that a reason for that? I have found it common in build scripts that I have seen

Reply:
@Bidski You know more about this stuff than I do, why do we use $() over ``

Reply:
[See this](http://mywiki.wooledge.org/BashFAQ/082)

Reply:
TL;DR `` are legacy and can do some weird things, $() is the new-fancy and does less weird things

Reply:
spicy! I'll keep that in mind